### ISSUE TITLE
Communicator not connecting when "Managed Stripping Level" is enabled

### ISSUE THREAD

**Issue by AcelisWeaven** [Unity-Technologies/ml-agents#2394]
**Describe the bug**
When "Managed Stripping Level" is set to Low/Medium/High (anything other than Disabled), the following error is shown when mlagents-learn is launched:
```
UnityAgentsException: The Communicator was unable to connect. Please make sure the External process is ready to accept communication with Unity.
  at MLAgents.Batcher.SendAcademyParameters (MLAgents.CommunicatorObjects.UnityRLInitializationOutput academyParameters) [0x0002d] in <e4d88f720ac843b581fe7c0a306443b6>:0
  at MLAgents.Academy.InitializeEnvironment () [0x00228] in <e4d88f720ac843b581fe7c0a306443b6>:0
  at MLAgents.Academy.Awake () [0x00000] in <e4d88f720ac843b581fe7c0a306443b6>:0
```

**To Reproduce**
1. Go to Project Settings > Player
2. Change the value of "Managed Stripping Level" to Low, Medium or High
3. Build your project and run the training

**Environment (please complete the following information):**
- OS + version: Windows 10, building for Linux
- _ML-Agents version_: ML-Agents v0.8
- _Environment_: Custom (may not be relevant)

**Comment by awjuliani** [comment]
Hi @AcelisWeaven 

Thanks for bringing this issue to our attention.

**Comment by chriselion** [comment]
I logged this as MLA-62 in our internal tracker. Are you OK with working around the problem in the meantime?

**Comment by AcelisWeaven** [comment]
Thanks! No worries 👍

**Comment by surfnerd** [comment]
We probably need to define a link.xml to make sure these symbols aren't stripped.  Docs [here](https://docs.unity3d.com/Manual/ManagedCodeStripping.html)

**Comment by surfnerd** [comment]
Hi @AcelisWeaven,
This issue has been fixed by #2662, which was just merged to the develop branch.  I am closing this issue, please reopen it if you still run into problems with code stripping in the future.
